modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CustomAuthorTableCell.tsx

```diff
@@ -8,7 +8,19 @@ import ClayLink from '@clayui/link';
 import {openModal} from 'frontend-js-web';
 import React from 'react';
 
-const SampleCustomDataRenderer = ({
+interface ICustomAuthorTableCell {
```

Comment:
What is the purpose of this interface? It's not exported and it's only used once inside the file so, IINM, it's not really enforcing anything...

Reply:
Code style choice, no strict purpose. My general thinking was that inline prop types are more readable when there are few of them (3-4 or below), and separate interface is more readable when there are a lot of props, like here. But that's not a strong opinion, in fact I can see value in avoiding naming quirks and code fragmentation. I wouldn't even be opposed to enforce no private interfaces rule in source formatter. I'll make the change.

Reply:
Mmmm :thinking: 

Reflecting about it I think that it's not the private interface itself what looks weird but the fact that it is only used once. If you just use it once then there's no point in declaring it. I mean, you could argue there's some value in the double accounting: interface and actual use (like what happens in TDD between production and test code), but I find it way too much (and I'm a type lover :sweat_smile: ).

If you used the interface at least once more in the same file (or expect to use it because chances are very high and the abstract definition has enough entity/makes sense) then I would go with the interface probably...